pkg/controller/clusters/management_test.go

```diff
@@ -0,0 +1,115 @@
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/controller/clientutil"
+	"github.com/aws/eks-anywhere/pkg/controller/clusters"
+)
+
+func TestFetchManagementEksaClusterSuccess(t *testing.T) {
+	tt := newFetchManagementTest(t)
+
+	objs := []runtime.Object{tt.cluster, tt.managementCluster}
+	cb := fakeClientBuilder()
+	cl := cb.WithRuntimeObjects(objs...).Build()
+
+	managementCluster, err := clusters.FetchManagementEksaCluster(context.Background(), cl, tt.cluster)
+	tt.Expect(err).To(BeNil())
+	tt.Expect(managementCluster).To(Equal(tt.managementCluster))
+}
+
+func TestFetchManagementEksaClusterDifferentNamespaceSuccess(t *testing.T) {
+	tt := newFetchManagementTest(t)
+	tt.managementCluster.Namespace = "different-namespace"
+
+	objs := []runtime.Object{tt.cluster, tt.managementCluster}
+	cb := fakeClientBuilder()
+	cl := cb.WithRuntimeObjects(objs...).Build()
+
+	managementCluster, err := clusters.FetchManagementEksaCluster(context.Background(), cl, tt.cluster)
+	tt.Expect(err).To(BeNil())
+	tt.Expect(managementCluster).To(Equal(tt.managementCluster))
+}
+
+func TestFetchManagementEksaClusterMissing(t *testing.T) {
+	tt := newFetchManagementTest(t)
+
+	tt.cluster.Spec.ManagementCluster.Name = "missing"
+	objs := []runtime.Object{tt.cluster, tt.managementCluster}
+	cb := fakeClientBuilder()
+	cl := cb.WithRuntimeObjects(objs...).Build()
+
+	managementCluster, err := clusters.FetchManagementEksaCluster(context.Background(), cl, tt.cluster)
+	tt.Expect(err.Error()).
+		To(Equal("unable to retrieve management cluster missing: clusters.anywhere.eks.amazonaws.com \"missing\" not found"))
+	tt.Expect(managementCluster).To(BeNil())
+}
+
+func TestFetchManagementEksaClusterMultiple(t *testing.T) {
+	tt := newFetchManagementTest(t)
+
+	tt.managementCluster.Namespace = "different-namespace-1"
+	managementCluster2 := tt.managementCluster.DeepCopy()
+	managementCluster2.Namespace = "different-namespace-2"
+	objs := []runtime.Object{tt.cluster, tt.managementCluster, managementCluster2}
+	cb := fakeClientBuilder()
+	cl := cb.WithRuntimeObjects(objs...).Build()
+
+	managementCluster, err := clusters.FetchManagementEksaCluster(context.Background(), cl, tt.cluster)
+	tt.Expect(err.Error()).To(Equal("found multiple clusters with the name my-management-cluster"))
+	tt.Expect(managementCluster).To(BeNil())
+}
+
+type fetchManagementTest struct {
+	*WithT
+	cluster           *anywherev1.Cluster
+	managementCluster *anywherev1.Cluster
+}
+
+func newFetchManagementTest(t *testing.T) *fetchManagementTest {
+	managementCluster, cluster := createClustersForTest()
+	cluster.SetManagedBy("my-management-cluster")
+	return &fetchManagementTest{
+		WithT:             NewWithT(t),
+		cluster:           cluster,
+		managementCluster: managementCluster,
+	}
+}
+
+func createClustersForTest() (managementCluster *anywherev1.Cluster, workloadCluster *anywherev1.Cluster) {
```

Comment:
nit, when I see `create` in the method it makes me think the method actually creates a cluster resource, but it's actually just returning a struct
```suggestion
func clustersForTest() (managementCluster *anywherev1.Cluster, workloadCluster *anywherev1.Cluster) {
```